jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java

```diff
@@ -549,8 +575,56 @@ public String getId()
      */
     void stop()
     {
-        mucs.values().forEach(MucWrapper::leave);
-        xmppConnection.disconnect();
+        synchronized(connectSynRoot)
+        {
+            mucs.values().forEach(MucWrapper::leave);
+            xmppConnection.disconnect();
+        }
+    }
+
+    /**
+     * The callable returned by this method describes the task of connecting
+     * and login to XMPP service.
+     *
+     * @param executorService - The scheduled executor service scheduling this
+     * callable. Once we are done reconnecting we can shut it down.
+     * @return <tt>Callable<Boolean></tt> which returns <tt>true</tt> as long
+     *         as we're failing to connect.
+     */
+    private Callable<Boolean> getConnectAndLoginCallable(
+        ScheduledExecutorService executorService)
+    {
+        return () ->
+        {
+            synchronized (connectSynRoot)
+            {
+                try
+                {
+                    xmppConnection.connect();
+                }
+                catch(Throwable t)
```

Comment:
why `Throwable` and not just an `Exception` here?

Reply:
Not a particular reason.

Reply:
If no reason then we're better off with using `Exception` for the reconnect try catch.